AppKit/CPTrackingArea.j

```diff
@@ -0,0 +1,141 @@
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+@import <Foundation/Foundation.j>
+@import "CPView.j"
+
+@typedef CPTrackingAreaOptions
+CPTrackingMouseEnteredAndExited     = 1 << 1,
+CPTrackingMouseMoved                = 1 << 2,
+CPTrackingCursorUpdate              = 1 << 3,
+CPTrackingActiveWhenFirstResponder  = 1 << 4,
+CPTrackingActiveInKeyWindow         = 1 << 5,
+CPTrackingActiveInActiveApp         = 1 << 6, // both mean the same in cappuccino context
+CPTrackingActiveAlways              = 1 << 6,
+CPTrackingAssumeInside              = 1 << 7,
+CPTrackingInVisibleRect             = 1 << 8,
+CPTrackingEnabledDuringMouseDrag    = 1 << 9;
+
+var CPTrackingAreaRectKey            = @"CPTrackinkAreaRectKey",
+    CPTrackingAreaOptionsKey         = @"CPTrackingAreaOptionsKey",
+    CPTrackingAreaOwnerKey           = @"CPTrackingAreaOwnerKey",
+    CPTrackingAreaUserInfoKey        = @"CPTrackingAreaUserInfoKey",
+    CPTrackingAreaReferencingViewKey = @"CPTrackingAreaReferencingViewKey";
+
+
+/*!
+ @ingroup appkit
+ 
+ A CPTrackingArea defines a region of view that generates mouse-tracking and
+ cursor-update events when the mouse is over that region.
+ */
+@implementation CPTrackingArea : CPObject
+{
+    CGRect                  _rect           @accessors(getter=rect);
+    CPTrackingAreaOptions   _options        @accessors(getter=options);
+    id                      _owner          @accessors(getter=owner);
+    CPDictionary            _userInfo       @accessors(getter=userInfo);
+    
+    CPView                  _referencingView;
+}
+
+
+#pragma mark -
+#pragma mark Initialization
+
+/*! Initializes and returns an object defining a region of a view to receive mouse-tracking events, mouse-moved events, cursor-update events, or possibly all these events.
+ */
+- (CPTrackingArea)initWithRect:(CGRect)aRect options:(CPTrackingAreaOptions)options owner:(id)owner userInfo:(CPDictionary)userInfo
+{
+    if (self = [super init])
+    {
+        if (!(options > 0))
+            [CPException raise:CPInternalInconsistencyException reason:"Invalid CPTrackingArea options"];
+
+        // Check options:
+        // - at least one in CPTrackingMouseEnteredAndExited, CPTrackingMouseMoved, CPTrackingCursorUpdate
+        // - exactly  one in CPTrackingActiveWhenFirstResponder, CPTrackingActiveInKeyWindow, CPTrackingActiveAlways
+        // - no check on CPTrackingAssumeInside, CPTrackingInVisibleRect, CPTrackingEnableDuringMouseDrag
+        
+        if (!((options & CPTrackingMouseEnteredAndExited) || (options & CPTrackingMouseMoved) || (options & CPTrackingCursorUpdate)))
+            [CPException raise:CPInternalInconsistencyException reason:"You must at least select a type of event message"];
+        
+        if ((((options & CPTrackingActiveWhenFirstResponder) > 0) + ((options & CPTrackingActiveInKeyWindow) > 0) + ((options & CPTrackingActiveAlways) > 0)) != 1)
```

Comment:
Yeah, that is a different test. I changed the code because Cocoa doesn't care if more than one is specified, but it does care if none are specified. Then we agreed it made sense to force only one as well.

So in addition to my "none" test, keep your "only one" test, but please change the message to:

"tracking area options may only specify one of [CPTrackingActiveWhenFirstResponder | CPTrackingActiveInKeyWindow | CPTrackingActiveInActiveApp | CPTrackingActiveAlways]."